include/bm/bluetooth/ble_scan.h

```diff
@@ -0,0 +1,570 @@
+	struct {
+		/** Short names that the main application will scan for, and that will be
+		 *  advertised by the peripherals.
+		 */
+		char short_target_name[CONFIG_BLE_SCAN_SHORT_NAME_MAX_LEN];
+		/** Minimum length of the short name. */
+		uint8_t short_name_min_len;
+	} short_name[CONFIG_BLE_SCAN_SHORT_NAME_COUNT];
+	/** Number of short target names. */
+	uint8_t name_cnt;
+	/** Flag to inform about enabling or disabling this filter. */
+	bool short_name_filter_enabled;
+};
+#endif
+
+#if (CONFIG_BLE_SCAN_ADDRESS_COUNT > 0)
+/** Scan address filter */
+struct ble_scan_addr_filter {
+	/** Addresses in the same format as the format used by the SoftDevice that the
+	 *  main application will scan for, and that will be advertised by the peripherals.
+	 */
+	ble_gap_addr_t target_addr[CONFIG_BLE_SCAN_ADDRESS_COUNT];
+	/** Number of target addresses. */
+	uint8_t addr_cnt;
+	/** Flag to inform about enabling or disabling this filter. */
+	bool addr_filter_enabled;
+};
+#endif
+
+#if (CONFIG_BLE_SCAN_UUID_COUNT > 0)
+/** Scan UUID filter */
+struct ble_scan_uuid_filter {
+	/** UUIDs that the main application will scan for, and that will be advertised by
+	 *  the peripherals.
+	 */
+	ble_uuid_t uuid[CONFIG_BLE_SCAN_UUID_COUNT];
+	/** Number of target UUIDs in list. */
+	uint8_t uuid_cnt;
+	/** Flag to inform about enabling or disabling this filter. */
+	bool uuid_filter_enabled;
+};
+#endif
+
+#if (CONFIG_BLE_SCAN_APPEARANCE_COUNT > 0)
+/** Scan appearance filter */
+struct ble_scan_appearance_filter {
+	/** Apperances that the main application will scan for, and that will be advertised by the
+	 *  peripherals.
+	 */
+	uint16_t appearance[CONFIG_BLE_SCAN_APPEARANCE_COUNT];
+	/** Number of appearances in list. */
+	uint8_t appearance_cnt;
+	/** Flag to inform about enabling or disabling this filter.*/
+	bool appearance_filter_enabled;
+};
+#endif
+
+/**
+ * @brief Filter data.
+ *
+ * @details This contains all filter data and the information about enabling and disabling
+ * any type of filters. Flag all_filter_mode informs about the filter mode. If this flag is set,
+ * then all types of enabled filters must be matched for the library to send a notification to the
+ * main application. Otherwise, it is enough to match one of the filters to send a notification.
+ */
+struct ble_scan_filters {
+#if (CONFIG_BLE_SCAN_NAME_COUNT > 0)
+	/** Name filter data. */
+	struct ble_scan_name_filter name_filter;
+#endif
+#if (CONFIG_BLE_SCAN_SHORT_NAME_COUNT > 0)
+	/** Short name filter data. */
+	struct ble_scan_short_name_filter short_name_filter;
+#endif
+#if (CONFIG_BLE_SCAN_ADDRESS_COUNT > 0)
+	/** Address filter data. */
+	struct ble_scan_addr_filter addr_filter;
+#endif
+#if (CONFIG_BLE_SCAN_UUID_COUNT > 0)
+	/** UUID filter data. */
+	struct ble_scan_uuid_filter uuid_filter;
+#endif
+#if (CONFIG_BLE_SCAN_APPEARANCE_COUNT > 0)
+	/** Appearance filter data. */
+	struct ble_scan_appearance_filter appearance_filter;
+#endif
```

Comment:
These members are not shown in API documentation because of the preprocessor conditionals. Give the configs a value through the `doc/nrf-bm/doxygen/nrf-bm.doxyfile` file?
There might be more hidden declarations. Please check.